SwanProjects/swanprojects/handlers.py

```diff
@@ -0,0 +1,289 @@
+
+        self.log.info(f"KernelSpecManagerPathHandler = {input_data}")
+
+        project = get_project_path(path)
+        if self.kernel_spec_manager.set_path(path):
+            data = {"status": True, "is_project": project is not None,
+                    "msg": f"SWAN kernel spec manager set to path: {path}"}
+            self.finish(json.dumps(data))
+        else:
+            data = {"status": False, "is_project": project is not None,
+                    "msg": f"Error setting SWAN kernel spec manager to path: {path}"}
+            self.finish(json.dumps(data))
+
+class CreateProjectHandler(APIHandler):
+    @tornado.web.authenticated
+    def post(self):
+        """
+        Endpoint to create a project, receive project information such as name, stack, platform, release, user_script.
+        The project is created at $HOME/SWAN_projects/project_name and a hidden json ".swanproject" file with the information
+        project is set inside the project folder.
+        """
+        input_data = self.get_json_body()
+        self.log.info(f"creating project {input_data}")
+
+        name = input_data["name"]
+        stack = input_data["stack"]  # CMSSW/LCG
+        platform = input_data["platform"]  # SCRAM/x86_64..centos7..gccX
+        release = input_data["release"]  # CMSSW_X_Y_Z/LCG_XYZ
+        user_script = input_data["user_script"]
+
+        project_dir = os.environ["HOME"] + "/SWAN_projects/" + name
+        try:
+            os.makedirs(project_dir)
+        except Exception as msg:
+            data = {"status": False, "project_dir": f"SWAN_projects/{name}",
+                    "msg": f"Error creating folder for project {name}, traceback: {msg}"}
+            self.finish(json.dumps(data))
+            return
+        swan_project_file = project_dir + os.path.sep + '.swanproject'
+        swan_project_content = {'stack': stack, 'release': release,
+                                'platform': platform}
+        try:
+            with open(swan_project_file, 'w+') as f:
+                f.write(json.dumps(swan_project_content,
+                        indent=4, sort_keys=True))
+                f.close()
+        except Exception as msg:
+            data = {"status": False, "project_dir": f"SWAN_projects/{name}",
+                    "msg": f"Error creating .swanproject file for project {name}, traceback: {msg}"}
+            self.finish(json.dumps(data))
+            return
+
+        try:
+            swan_user_script_file = project_dir + os.path.sep + '.userscript'
+            with open(swan_user_script_file, 'w') as f:
+                f.write(user_script)
+                f.close()
+        except Exception as msg:
+            data = {"status": False, "project_dir": f"SWAN_projects/{name}",
+                    "msg": f"Error creating .userscript file for project {name}, traceback: {msg}"}
+            self.finish(json.dumps(data))
+            return
+
+        command = get_env_isolated()
```

Comment:
not sure if we could avoid calling a python script from bash instead of just using its functions directly.